eclair-core/src/main/scala/fr/acinq/eclair/tor/Socks5Connection.scala

```diff
@@ -237,6 +237,11 @@ object Socks5ProxyParams {
       case _: IPv6 if proxyParams.useForIPv6 => Some(proxyParams.address)
       case _: Tor2 if proxyParams.useForTor => Some(proxyParams.address)
       case _: Tor3 if proxyParams.useForTor => Some(proxyParams.address)
+      case _: DnsHostname => InetAddress.getByName(address.host) match {
```

Comment:
Good point!  `proxyAddress()` should only decide whether or not to use a proxy for a particular address type.

I also agree it's better to add  a parameter that makes it explicit if a socks5 proxy should be used for DNS hostnames. I prefer `proxyParams.useForDnsHostnames` instead of `useForDnsHosts` because it better matches the internal naming if someone searches the code. The corresponding config file parameter is then `socks5.use-for-dnshostnames` .

Note: DNS hostnames are resolved as `remoteAddress` in Client.scala. After connecting to a proxy a `Socks5Connection` uses `remoteAddress` and expects it to be a resolved ipv4 or ipv6 address. 

fixed in [ec79f2e](https://github.com/ACINQ/eclair/pull/2234/commits/ec79f2e581f34af9a41356c0aade3e5e7640b4ed)